test/fika/parser_test.exs

```diff
@@ -61,8 +84,51 @@ defmodule Fika.ParserTest do
       assert {:ok, {3.0, "Float", _}} = TE.eval("1 * 5 - 4 / 2")
     end
 
-    test "boolean & has less precedence than !" do
-      assert {:ok, {false, "Bool", _}} = TE.eval("!true & :false")
+    test "comparison operators <, >, <= and >= have less precedence than arithmetic + and -" do
+      assert {:ok, {false, "Bool", _}} = TE.eval("1 + 2 < 4 - 1")
+      assert {:ok, {true, "Bool", _}} = TE.eval("1 + 2 <= 4 - 1")
+      assert {:ok, {false, "Bool", _}} = TE.eval("5 - 2 > 2 + 1")
+      assert {:ok, {true, "Bool", _}} = TE.eval("5 - 2 >= 2 + 1")
+    end
+
+    # TODO: Test using evaluation instead of AST as soon as == and != support booleans.
+    #   Replace the content of this test with something like:
+    #     assert {:ok, {true, "Bool", _}} = TE.eval("1 < 1 == 2 > 2")
+    #     assert {:ok, {true, "Bool", _}} = TE.eval("1 <= 1 == 2 >= 2")
+    test "comparison operators == and != have less precedence than <, >, <=, >=" do
```

Comment:
Due to the fact that for now we are implementing comparison operator for integers only, I cannot properly test that `==` and `!=` have higher precedence than `<`, `>`, `<=`, `>=` (using evaluation I mean).
So I left some example in the comments as a guide for when we support other types.
As of now, I'm testing that the parser produces the right AST and then that `==` and `!=` work with simple integers.